torchdynamo/config.py

```diff
@@ -66,3 +66,12 @@
     "torch.distributions",
     "torch.testing",
 }
+
+# Enable this check if you expect your eval model to change on forward passes - this ensures guards do a deeper check on parameter names and any named parameters you expect to change (see: eval_model_named_parameter_checks).
+# Note: guard_nn_modules must be True for this to run.
+eval_model_named_parameter_checks_enabled = True
```

Comment:
How expensive is this?  (You can check with torchbench.py --overhead.)

I'd be tempted just to turn this on for everything (and all names) if the cost is low.